### DISCUSSION TITLE
Wrapping a HStack Causes Leading Space

### DISCUSSION THREAD

**Discussion by st11x** [chakra-ui/chakra-ui#5900] [Q&A] [answered]
Hi,

I have a HStack that I applied the wrap property to as in this code

```
    <HStack wrap={{ base: "wrap", md: "nowrap" }}>
    <FormControl>
        <FormLabel htmlFor="startdate">Start Date</FormLabel>
        <Input id="startdate" type="date"></Input>
    </FormControl>
    <FormControl>
        <FormLabel htmlFor="enddate">End Date</FormLabel>
        <Input id="enddate" type="date"></Input>
    </FormControl>
    <FormControl>
        <FormLabel htmlFor="Position">Position</FormLabel>
        <Select>
            <option>v</option>
        </Select>
    </FormControl>
    </HStack>
```

This is at size md and above, no issues here:

![image](https://user-images.githubusercontent.com/7398499/164074103-7cd38979-bade-44c9-9b0e-fc1068032081.png)

at a smaller view port, I see these leading spaces for the wrapped item.

![image](https://user-images.githubusercontent.com/7398499/164074269-6fee8270-6bd5-4d51-b769-a7c7ff55ab79.png)

I can fix the wrap by swapping Wrap for HStack, but then I couldn't tell it not to wrap at md and above. 

Why is there a leading space? 

thanks
  Matt

**Comment by eckertalex** [comment, marked as answer]
`HStack` is a horizontal stack. It automatically inserts a space between each element. With the `wrap` prop you force the stack to wrap when there is not enough horizontal space. However, the space between the elements is still there.

I created a [CodeSandbox](https://codesandbox.io/s/chakra-discussion-wrapping-hstack-sx4vou?file=%2Fsrc%2Fapp.tsx) with my recommended solution for this. Use a `Stack` and change the direction smaller viewports.

**Reply by st11x** [reply]
Thanks for the solution. It works when you add a gap prop too, so that adds the spacing back between the elements.
